subsys/net/lib/download_client/src/http.c

```diff
@@ -40,6 +40,8 @@ LOG_MODULE_DECLARE(download_client, CONFIG_DOWNLOAD_CLIENT_LOG_LEVEL);
 	"Connection: keep-alive\r\n"                                           \
 	"\r\n"
 
+extern char *strnstr(const char *haystack, const char *needle, size_t haystack_sz);
```

Comment:
I recall there was some reson, but please remind me why can't we simply include the respective header?

Reply:
I didn't dig much but as far as I understood, it's defined in `string.h` but behind `__BSD_VISIBLE` (in `picolibc`) because it's a non-standard BSD extension of the C library. That header is already included and I got a warning but it linked. So I simply declared it there.

Reply:
It's there by default with both `picolibc` and `newlibc-nano`.

Reply:
@lemrey can this information be added as a comment on top of the `extern` for future reference?